Source/ASCollectionNode.mm

```diff
@@ -193,18 +193,27 @@ - (void)clearContents
   [self.rangeController clearContents];
 }
 
-- (void)didExitPreloadState
-{
-  [super didExitPreloadState];
-  [self.rangeController clearPreloadedData];
-}
-
 - (void)interfaceStateDidChange:(ASInterfaceState)newState fromState:(ASInterfaceState)oldState
 {
   [super interfaceStateDidChange:newState fromState:oldState];
   [ASRangeController layoutDebugOverlayIfNeeded];
 }
 
+- (void)didEnterPreloadState
+{
+  // Intentionally allocate the collection view here so that super will trigger a layout pass on it which in turn will loads its intial data.
+  // We can get rid of this later when ASDataController, ASRangeController and the collection layout object can operate without the view.
+  [self view];
+  [super didEnterPreloadState];
+}
+
+- (void)didEnterDisplayState
+{
+  [super didEnterDisplayState];
+  // Since an initial data load was triggered when this node enter preloads state, wait for it to finish
+  [self waitUntilAllUpdatesAreCommitted];
```

Comment:
This is the scariest part of the change, @Adlai-Holler can you double check that this makes sense to you as well?